src/components/wizard-upload-configfile/wizard-upload-configfile.js

```diff
@@ -0,0 +1,47 @@
+import {copy, fromJson, module} from 'angular';
+
+export default module('app.components.wizard-upload-configfile', [])
+  .component('wizardUploadConfigfile', {
+    bindings: {
+      detailsString: '@',
+      onUpdate: '&',
+      fileContent: '<',
+    },
+    controller: class WizardTunnelCtrl {
+      constructor($scope) {
+        'ngInject';
+
+        this.success = false;
+        this.$scope = $scope;
+        this.$ctrl = $scope.$ctrl;
+
+        $scope.$watch('$ctrl.fileContent', this.updateFromInput.bind(this));
+        $scope.$watch('$ctrl.newFileContent', this.updateOutput.bind(this));
+      }
+
+      $onInit() {
```

Comment:
This should work... what's going wrong if you pass the object? Undefined value? The JSON stringify and parse operations make it overly complex right now.